src/PiTop.Algorithms/VelocitySensor.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Reactive;
+using System.Reactive.Concurrency;
+using System.Reactive.Linq;
+using UnitsNet;
+
+namespace PiTop.Algorithms
+{
+    public class VelocitySensor : IDisposable
+    {
+        private readonly IDisposable _sampler;
+        private (DateTimeOffset Timestamp, Acceleration Accelecation)? _previousSample;
+        private IScheduler _scheduler;
+
+        public VelocitySensor(TimeSpan samplingFrequency, Func<Acceleration> accelerationSampler, IScheduler samplingScheduler = null)
+        {
+            Velocity = Speed.Zero;
+            _scheduler = samplingScheduler ?? TaskPoolScheduler.Default;
+            var samplingFrequency1 = samplingFrequency;
+            if (samplingFrequency1.TotalMilliseconds <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(samplingFrequency), "Sampling frequency must be greater than 0 ms");
+            }
+            var accelerationSampler1 = accelerationSampler ?? throw new ArgumentNullException(nameof(accelerationSampler));
+            _sampler = Observable
+                .Interval(samplingFrequency1, _scheduler)
+                .Select(_ => (_scheduler.Now, accelerationSampler1()))
+                .Subscribe(OnNext);
```

Comment:
We should document that we assume that the time taken do sample acceleration is negligible compared to the sample interval. Alternative assumptions are possible for more accuracy when a high sample rate is desired, but this is the easiest one. Let's stick with that since we're not on an RTOS anyway, so high sample rates and accuracy don't go together really well.

Reply:
code has changed to include the acceleration sampling time